Sources/WhoopDIKit/Module/AccumulationDefinition.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+
+private func wrapAccumulationProvider<Key: AccumulationKey>(name: String?, accumulationKey: Key.Type, valueProvider: @escaping (Any?) throws -> Key.AccumulatedValue) -> (Any?, Container) throws -> Any {
+    return { (params: Any?, container: Container) throws -> Any in
+        // Accumulate from the parent value if it exists, then add the current one
+        let previousValueFromContainer: Key.FinalValue?
+        do {
+            if let parent = container.parent {
+                let value: Key.FinalValue = try parent.get(name, params)
+                previousValueFromContainer = value
+            } else {
+                previousValueFromContainer = nil
+            }
+        } catch DependencyError.missingDependency {
+            previousValueFromContainer = nil // If there is no dependency, make it nil. Fail on other errors
+        }
+        return accumulationKey.accumulate(current: previousValueFromContainer ?? accumulationKey.defaultValue,
```

Comment:
This can lead to a stackoverflow if we have more than a certain amount of values we are trying to accumulate (2^14 in the simulator). This seems like a large amount but, it's easier to hit if this is nested as a dependency of a dependency.